perun-registrar-lib/src/test/java/cz/metacentrum/perun/registrar/RegistrarBaseIntegrationTest.java

```diff
@@ -794,6 +799,291 @@ private Application prepareApplicationToGroup(User user, Group group) {
+	}
+
+	@Test
+	public void getApplicationsPageFindByUsername() throws Exception {
+		System.out.println("getApplicationsPageFindByGroupName");
+
+		// create groups
+		Group group1 = setUpGroup("Group1", "Cool folks");
+		Group group2 = setUpGroup("Group2", "Cooler folks");
+		// create users
+		User user1 = setUpUser("Joe", "Doe");
+		User user2 = setUpUser("Barney", "Stinson");
+
+		setUpApplicationGroup(user1, group1);
+		setUpApplicationGroup(user2, group2);
+
+		ApplicationsPageQuery query = new ApplicationsPageQuery(4, 0, SortingOrder.DESCENDING, ApplicationsOrderColumn.ID, user2.getLastName(), List.of("APPROVED", "VERIFIED"), true);
+
+		Paginated<Application> result = registrarManager.getApplicationsPage(session, vo, query);
+
+		List<User> returnedUsers = result.getData().stream()
+			.map(Application::getUser).toList();
+
+		List<String> returnedUserLastNames = returnedUsers.stream()
+				.map(User::getLastName).toList();
+
+		assertThat(returnedUserLastNames).containsOnly(user2.getLastName());
+	}
+
+	@Test
+	public void getApplicationsPageFindByGroupName() throws Exception {
+		System.out.println("getApplicationsPageFindByGroupName");
+
+		// create groups
+		Group group1 = setUpGroup("Group1", "Cool folks");
+		Group group2 = setUpGroup("Group2", "Cooler folks");
+		// create users
+		User user1 = setUpUser("Joe", "Doe");
+		User user2 = setUpUser("Barney", "Stinson");
+
+		setUpApplicationGroup(user1, group1);
+		setUpApplicationGroup(user2, group2);
+
+		ApplicationsPageQuery query = new ApplicationsPageQuery(4, 0, SortingOrder.DESCENDING, ApplicationsOrderColumn.ID, "Group2", List.of("APPROVED", "VERIFIED"), true);
+
+		Paginated<Application> result = registrarManager.getApplicationsPage(session, vo, query);
+
+		List<Group> returnedGroups = result.getData().stream()
+			.map(Application::getGroup).filter(Predicate.not(Objects::isNull)).toList();
+
+		assertThat(returnedGroups).containsOnly(group2);
+	}
+
+	@Test
+	public void getApplicationsPageFindByGroupDescription() throws Exception {
+		System.out.println("getApplicationsPageFindByGroupDescription");
+
+		// create groups
+		Group group1 = setUpGroup("Group1", "Cool folks");
+		Group group2 = setUpGroup("Group2", "Cooler folks");
+		// create users
+		User user1 = setUpUser("Joe", "Doe");
+		User user2 = setUpUser("Barney", "Stinson");
+
+		setUpApplicationGroup(user1, group1);
+		setUpApplicationGroup(user2, group2);
+
+		ApplicationsPageQuery query = new ApplicationsPageQuery(4, 0, SortingOrder.DESCENDING, ApplicationsOrderColumn.ID, "cooler", List.of("APPROVED", "VERIFIED"), true);
+
+		Paginated<Application> result = registrarManager.getApplicationsPage(session, vo, query);
+
+		List<Group> returnedGroups = result.getData().stream()
+			.map(Application::getGroup).filter(Predicate.not(Objects::isNull)).toList();
+
+		assertThat(returnedGroups).containsOnly(group2);
+	}
+
+	@Test
+	public void getApplicationsPageFindByGroupId() throws Exception {
```

Comment:
Could you also add tests where groupId is not used as search string, but as groupId variable in the query and the same for the memberId?